### PR TITLE
Add webhook check for PVC modification

### PR DESCRIPTION
**Pull request by anyasabo**
Partially resolves https://github.com/elastic/cloud-on-k8s/issues/1293

Adds webhook validation for changes to any existing nodes (node set? node group? 😄 )

We may want to validate in the operator too. If for some reason the webhook is not deployed, we try to keep updating the sset and getting rejected. I'm not sure if that's a use case we really need to worry about though.

I also added the breaking label to this but not sure since it's really the sset change that is breaking, this just surfaces it.

### CODE REVIEW

File: operators/pkg/controller/elasticsearch/validation/validation.go [hunk 1]
```diff
@@ -81,7 +82,7 @@ func Validate(es estype.Elasticsearch) ([]validation.Result, error) {
 			Version:       *v,
 		},
 	}
-	var errs []validation.Result
+	var errs []validation.Result //nolint:go-lint
```
Comment (david-kow) [inline operators/pkg/controller/elasticsearch/validation/validation.go]: Is nolint needed?

File: operators/pkg/controller/elasticsearch/validation/validations.go [hunk 2]
```diff
@@ -149,3 +151,36 @@ func tlsCannotBeDisabled(ctx Context) validation.Result {
...
+// pvcModification ensures no PVCs are changed, as volume claim templates are immutable in stateful sets
+func pvcModification(ctx Context) validation.Result {
+	if ctx.Current == nil {
+		return validation.OK
+	}
+	for _, node := range ctx.Proposed.Elasticsearch.Spec.Nodes {
+		currNode := getNode(node.Name, ctx.Current.Elasticsearch)
+		if currNode == nil {
+			// this is a new sset, so there it is okay
+			return validation.OK
```
Comment (david-kow) [inline operators/pkg/controller/elasticsearch/validation/validations.go]: What will happen if first sset is new and the next one is already existing but with modified volume claim templates?

File: operators/pkg/controller/elasticsearch/validation/validation.go (hunk 1 above)
Comment (anyasabo) [reply operators/pkg/controller/elasticsearch/validation/validation.go]: Yeah it complained about not preallocating, though maybe there's a difference in configuration in my editor linter and the one we use for testing. Might need to reconcile that on my side. I'll remove it here if needed

File: operators/pkg/controller/elasticsearch/validation/validations.go (hunk 2 above)
Comment (anyasabo) [reply operators/pkg/controller/elasticsearch/validation/validations.go]: A bug :)

I'll fix

**Comment by anyasabo** [comment]
>All validations are also run in the main reconciliation loop in case the webhook is not deployed or fails, so all we have to do is maybe improve the visibility of the validation results because they are currently only in the logs.

When I was doing some manual testing I thought I noticed that was not the case, I'll double check

**Comment by pebrc** [comment]
Actually it should update the status of the Elasticsearch resource, but that could be buggy, we should double check

https://github.com/elastic/cloud-on-k8s/blob/58b28fb3a233b0b0ff96ca0c9248b59bd233aadf/operators/pkg/controller/elasticsearch/elasticsearch_controller.go#L249-L256

**Comment by anyasabo** [comment]
@pebrc thanks for pointing me in the right direction. I verified it was working as expected for other validations. It does not work for this validation because this relies on the `current` (vs `proposed`) ES spec. Since that is called inside of the reconcile loop, there is no difference as the ES object has already made it through admission. We would need to add code to check the current ES spec and the current sset specs to achieve something similar outside of the webhook.

**Pull request merged**